RTClib.cpp

```diff
@@ -415,6 +454,65 @@ bool DateTime::isValid() const {
          mm == other.mm && ss == other.ss;
 }
 
+/**************************************************************************/
+/*!
+    @author Harrison Outram
+    @brief  Fixes DateTime object if invalid
+
+    Determines if any date or time components are too high.
+    E.g. seconds == 65
+
+    Increments next component and reduces invalid component to fix.
+    E.g. if seconds == 125, then minutes goes up by 2 and seconds
+    goes down to 5.
+
+    Does nothing if the DateTime object is already valid
+
+    @warning Will still result in invalid DateTime object if year is above 2099
+    @return true if fixed, false if year becomes invalid
+*/
+/**************************************************************************/
+bool DateTime::fixDateTime() {
+  uint8_t temp;
+
+  if (ss >= 60) {
+    temp = ss / 60;
+    mm += temp;
+    ss -= 60 * temp;
+  }
+  if (mm >= 60) {
+    temp = mm / 60;
+    hh += temp;
+    mm -= 60 * temp;
+  }
+  if (hh >= 24) {
+    temp = hh / 24;
+    d += temp;
+    hh -= temp * 24;
+  }
+
+  // make month valid to prevent getDaysInMonth() returning 0
+  // Otherwise, infinite loop possible
+  if (m > 12) {
+    temp = m % 12;
+    yOff += temp;
+    m -= temp * 12;
+  }
+
+  temp = getDaysInMonth(yOff, m);
+  while (d > temp) {
+    d -= temp;
+    m++;
+    if (m > 12) {
+      yOff++;
+      m--;
```

Comment:
This should be `m = 1` or `m -= 12`
```suggestion
      m -= 12;
```

Reply:
fixed